build/media_source/system/js/multiselect.es6.js

```diff
@@ -6,108 +6,117 @@
-          // We will prevent selecting text to prevent artifacts
-          if (shiftKey) {
-            document.body.style['-webkit-user-select'] = 'none';
-            document.body.style['-moz-user-select'] = 'none';
-            document.body.style['-ms-user-select'] = 'none';
-            document.body.style['user-select'] = 'none';
-          }
+    // Select rows in range
+    if (shiftKey && this.prevRow) {
+      // Prevent text selection
+      document.getSelection().removeAllRanges();
 
-          currentCheckBox.checked = !currentCheckBox.checked;
-          isChecked = currentCheckBox.checked;
-          Joomla.isChecked(isChecked, this.tableEl.id);
-        }
+      // Re-query all rows, because they may be modified during sort operations
+      const rows = this.getRows();
+      const idxStart = rows.indexOf(this.prevRow);
+      const idxEnd = rows.indexOf(currentRow);
 
-        this.changeBg(this.rows[currentRowNum], isChecked);
+      // Check for more than 2 row selected
+      if (idxStart >= 0 && idxEnd >= 0 && Math.abs(idxStart - idxEnd) > 1) {
+        const slice = idxStart < idxEnd ? rows.slice(idxStart, idxEnd + 1) : rows.slice(idxEnd, idxStart + 1);
 
-        // Restore normality
-        if (shiftKey) {
-          document.body.style['-webkit-user-select'] = 'none';
-          document.body.style['-moz-user-select'] = 'none';
-          document.body.style['-ms-user-select'] = 'none';
-          document.body.style['user-select'] = 'none';
-        }
+        slice.forEach((row) => {
+          if (row === currentRow) {
+            return;
+          }
+          const rowBox = row.querySelector(this.boxSelector);
+          if (rowBox && rowBox.checked !== isChecked) {
+            rowBox.checked = isChecked;
+            this.changeBg(row, isChecked);
+            Joomla.isChecked(isChecked, this.formEl);
+          }
+        });
       }
     }
+
+    this.prevRow = currentRow;
+  }
+}
+
+const onBoot = (container) => {
+  let selector = '#adminForm';
+  const confSelector = window.Joomla ? Joomla.getOptions('js-multiselect', {}).formName : '';
+
+  if (confSelector) {
+    const pref = confSelector[0];
+    selector = (pref !== '.' && pref !== '#') ? `#${confSelector}` : confSelector;
   }
 
-  const onBoot = () => {
-    let formId = '#adminForm';
-    if (Joomla && Joomla.getOptions('js-multiselect', {}).formName) {
-      formId = `#${Joomla.getOptions('js-multiselect', {}).formName}`;
+  container.querySelectorAll(selector).forEach((formElement) => {
+    if (formElement && !('multiselect' in formElement.dataset)) {
+      formElement.dataset.multiselect = '';
+      // eslint-disable-next-line no-new
+      new JMultiSelect(formElement);
     }
-    // eslint-disable-next-line no-new
-    new JMultiSelect(formId);
-  };
+  });
+};
 
-  document.addEventListener('DOMContentLoaded', onBoot);
-})(Joomla);
+document.addEventListener('DOMContentLoaded', () => onBoot(document));
```

Comment:
@Fedik since this is a `type=module` you don't need the event here:

```suggestion
onBoot(document);
```

Reply:
Yeap, but I will keep it ;)

Reply:
The `DOMContentLoaded` event is a performance issue and it's best to not be used anymore...
That goes for all the JS scripts...

Reply:
It just a listener, there still a timing diference betwen use or not use it.
For this script is fine as it is.

Reply:
Okay, it does not hurt anyway :)